src/domainobjects/generatable.py

```diff
@@ -39,90 +186,391 @@ def generate_random_string(self, length,
+
+        Returns
+        -------
+        int
+            Integer between given min, max values, or of given length, and
+            negative if specified.
+        """
+
         if length is not None:
             min = 10**(length-1)
             max = (10**length)-1
 
         value = random.randint(min, max)
         return value if not negative else -value
-    
+
     def generate_random_decimal(self, min=10, max=10000, dp=2):
+        """ Generate a random number between a given range to a given number
+        of decimal places.
+
+        Parameters
+        ----------
+        min : int
+            Minimum value of the range
+        max : int
+            Maximum value of the range
+        dp : int
+            Number of decimal places to generate the value to
+
+        Returns
+        -------
+        float
+            Randomly generated value between min and max to dp decimal places
+        """
+
         return round(random.uniform(min, max), dp)
 
     def generate_currency(self):
+        """ Generate a random currency from a set list
+
+        Returns
+        -------
+        String
+            Random currency from a pre-defined list
+        """
+
         return random.choice(self.CURRENCIES)
-    
+
     def generate_asset_class(self):
+        """ Generate a random asset class from a set list
+
+        Returns
+        -------
+        String
+            Random asset class from a pre-defined list
+        """
+
         return random.choice(self.ASSET_CLASSES)
-    
+
     def generate_ric(self, ticker, exchange_code):
+        """ Appends two input values to "ticker.exchange_code"
+
+        Parameters
+        ----------
+        ticker : String
+            a ticker
+        exchange_code : String
+            an exchange code
+
+        Returns
+        -------
+        String
+            Combined "ticker.exchange_code"
+        """
+
         return '{0}.{1}'.format(ticker, exchange_code)
-    
+
     def generate_isin(self, coi, cusip):
-        return ''.join([coi, cusip, '4'])
-    
+        """ Appends two input values to "'coi''cusip''4'"
+
+        Parameters
+        ----------
+        coi : String
+            a country of issuance code
+        cusip : int
+            a CUSIP value
+
+        Returns
+        -------
+        String
+
+        """
+        return ''.join([coi, str(cusip), '4'])
+
     def generate_credit_debit(self):
+        """ Generate a random credit or debit value
+
+        Returns
+        -------
+        String
+            Random value between credit or debit
+        """
+
         return random.choice(self.CREDIT_DEBIT)
-    
+
     def generate_long_short(self):
+        """ Generate a random long or short value
+
+        Returns
+        -------
+        String
+            Random value between long or short
+        """
+
         return random.choice(self.LONG_SHORT)
-    
+
     def generate_position_type(self, no_sd=False, no_td=False):
```

Comment:
Why doesn't this work like the others where there is a default list of POSITION_TYPES = [TD, SD] and there is the option of overwritting that list ?